core/src/main/java/org/apache/iceberg/FileCleanupStrategy.java

```diff
@@ -83,4 +85,25 @@ protected void deleteFiles(Set<String> pathsToDelete, String fileType) {
             (file, thrown) -> LOG.warn("Delete failed for {} file: {}", fileType, file, thrown))
         .run(deleteFunc::accept);
   }
+
+  protected Set<String> expiredStatisticsFilesLocations(
+      TableMetadata beforeExpiration, TableMetadata afterExpiration) {
+    Set<String> statisticsFilesLocationBeforeExpiry = allStatisticsFilesLocation(beforeExpiration);
+    Set<String> statisticsFilesLocationAfterExpiry = allStatisticsFilesLocation(afterExpiration);
+    statisticsFilesLocationBeforeExpiry.removeAll(statisticsFilesLocationAfterExpiry);
+    return statisticsFilesLocationBeforeExpiry;
```

Comment:
I think it is a best practice to avoid mutating sets or returning mutable sets. Can you use Guava's `Sets.difference` instead of `removeAll`? It would also be better if the set returned by `allStatisticsFilesLocations` were immutable.